src/recensio/plone/browser/publications.py

```diff
@@ -9,6 +10,34 @@ def _render_cachekey(method, self, brain, lang):
     return (brain.getPath(), lang, DateTime().dayOfYear())
 
 
+class PublicationDefaultPage(DefaultPage):
+    """Translated default page"""
+
+    def getDefaultPage(self):
+        """Get the default page, if possible in the correct language.
+        We expect the default page to have no language suffix (e.g. 'index').
+        Translations should have a language suffix (e.g. 'index-en').
+        If this fails we check the language attribute of all contained documents.
+        If there is only one document with the right language we return that.
+        """
+        default_page = super().getDefaultPage()
+        language = api.portal.get_current_language()
+        defob = self.context.get(default_page)
+        if not defob or defob.Language() == language:
+            return default_page
+        documents = self.context.listFolderContents(dict(portal_type="Document"))
+        id_match = [
+            item for item in documents if item.getId() == f"{default_page}-{language}"
+        ]
+        if len(id_match) == 1:
+            return id_match[0].getId()
+
+        language_match = [item for item in documents if item.Language() == language]
+        if len(language_match) == 1:
```

Comment:
IMO we could also return just the first matched document, is we retrieve them in the correct sort order.
That way you could implicitly define a default page via it's objectPositionInParent, something I favor over default pages and made an addon in: https://github.com/collective/collective.folderishtraverse/

But this would probably lead to troubles in the context of recensio.

So, LGTM.